app/assets/javascripts/sequence_show.js

```diff
@@ -105,9 +131,255 @@ function initSequenceShow(data) {
         }
     }
 
+    function setUpFA(fa) {
+        setUpGO(fa.go);
+        setUpEC(fa.ec);
+    }
+
+    function setUpEC({numAnnotatedPeptides, data}) {
+        let ecResultset = new ECNumbers({numAnnotatedPeptides, data});
+        if (numAnnotatedPeptides > 0) {
+            setUpECTree(ecResultset);
+            setUpEcTable(ecResultset);
+
+            $(".ecNumberLink")
+                .mouseenter(function (e) {
+                    let ecNum = this.textContent;
+                    $tooltip.html(tooltipEC(ecNum, ecResultset));
+                    return false;
+                });
+        } else {
+            $("#ec-table").html("<span>No EC code annotations found.</span>");
+            $("#ec-treeview").remove();
+        }
+    }
+
+    /**
+     * Generate a tooltip for an EC number
+     * @param  {string}    ecNumber   The Ec number to generate a tooltip for
+     * @param  {ECNumbers} [ecResultSet=null]  A `ECNumbers` summary
+     * @return {string}    HTML for the tooltip
+     */
+    function tooltipEC(ecNumber, ecResultSet = null) {
+        const fmt= x=>`<div class="tooltip-ec-ancestor"><span class="tooltip-ec-term">EC ${x}</span><span class="tooltip-ec-name">${ECNumbers.nameOf(x)}</span></div>`;
+
+        let result = `
+            <h4 class="tooltip-fa-title">
+                <span class="tooltip-fa-title-name">${ECNumbers.nameOf(ecNumber)}</span>
+                <span class="tooltip-ec-term small"> EC ${ecNumber}</span>
+            </h4>`;
+
+        if (ECNumbers.ancestorsOf(ecNumber).length > 0) {
+            result += `${ECNumbers.ancestorsOf(ecNumber).map(c => fmt(c)).join("\n")}`;
+        }
+
+        if (ecResultSet != null) {
+            let count = ecResultSet.getValueOf(ecNumber);
+            result += `<div class="tooltip-fa-text">Specificly assigned to ${numberToPercent(ecResultSet.getFractionOf(ecNumber), 1)} (${count}) of
+            annotated matches</div>`;
+        }
+        return result;
+    }
+
     /**
-     * Inits the lineage tree
+     * Create the EC treeview
+     *
+     * @param {ECNumbers} ecResultSet  A `ECNumbers` summary
+     * @return {TreeView} The created treeview
      */
+    function setUpECTree(ecResultSet) {
+        let tree = ecResultSet.createTree("#ec-treeview", {
+            width: 916,
```

Comment:
magic numbers

_[Original comment](https://github.ugent.be/unipept/unipept/pull/625#discussion_r13724) by @bmesuere on Mon Feb 19 2018 at 14:27._